lib/models/event.dart

```diff
@@ -55,160 +47,76 @@ class Event {
-  } catch (e) {
-    throw Exception("Given map is malformed!\n$e");
-  }
-}
-
-  Event.clone(Event e): this(
-    name: e.name, 
-    id: e.id, 
-    description: e.description, 
-    color: e.color, 
-    location: e.location, 
-    tags: e.tags, 
-    recurrenceRules: e.recurrenceRules,
-    timeStart: e.timeStart, 
-    timeEnd: e.timeEnd, 
-    timeCreated: e.timeCreated, 
-    timeModified: e.timeModified);
-
-  Map<String, dynamic> toMap() {
-    return ({
-      'date created': timeCreated,
-      'date modified': timeModified,
-      'description': description,
-      'event time start': timeStart,
-      'event time end': timeEnd,
-      'hex color': color,
-      'location': location,
-      'recurrence rules': recurrenceRules?.toMap(),
-      'tags': tags.toList(),
-      'event name': name
-    });
-  }
-
-  set name(String newName) {
-    _timeModified = DateTime.now();
-    _name = newName;
-  }
-
-  String get name => _name;
-  
-  set id(String newId) {
-    _timeModified = DateTime.now();
-    _id = newId;
-  }
-
-  String get id => _id;
-
-  set description(String newDescription) {
-    _timeModified = DateTime.now();
-    _description = newDescription;
-  }
-
-  String get description => _description;
-
-  set color(String newColor) {
-    _timeModified = DateTime.now();
-    _color = newColor;
-  }
-
-  String get color => _color;
-
-  set location(String newLocation) {
-    _timeModified = DateTime.now();
-    _location = newLocation;
+  /// Turn a properly formatted map into an Event class
+  /// the map must have all the proper fields
+  Event.fromMap(Map<String, dynamic> map, {String? id})
+      : super.fromMap(map, id: id) {
+    _timeEnd = toDateIfTimestamp(map["time end"]);
   }
 
-  String get location => _location;
-
-  set tags(List<String> newTags) {
-    _timeModified = DateTime.now();
-    _tags = newTags;
-  }
-
-  List<String> get tags => _tags;
-
-  set recurrenceRules(Recurrence newRecurrence) {
-    _timeModified = DateTime.now();
-    _recurrenceRules = newRecurrence;
-  }
-
-  Recurrence get recurrenceRules => _recurrenceRules;
-
-  set timeStart(DateTime newTimeStart) {
-    _timeModified = DateTime.now();
-    _timeStart = newTimeStart;
+  Event.clone(Event e)
+      : this(
+            name: e.name,
+            id: e.id,
+            description: e.description,
+            color: e.color,
+            location: e.location,
+            tags: e.tags,
+            recurrenceRules: e.recurrenceRules,
+            timeStart: e.timeStart,
+            timeEnd: e.timeEnd,
+            timeCreated: e.timeCreated,
+            timeModified: e.timeModified);
+
+  /// returns a mapping with kv pairs corresponding to Firebase's
+  /// possibly a better getter
+  @override
+  Map<String, dynamic> toMap({keepClasses = false, includeID = false}) {
+    Map<String, dynamic> map = super.toMap();
+    map['time end'] = timeEnd;
+    return map;
```

Comment:
Not in superclass because tasks do not need cloning as of now (and for the forseeable future)